### ISSUE TITLE
Using authentication

### ISSUE THREAD

**Issue by konstantinmiller** [ClearcodeHQ/pytest-elasticsearch#327]
That's not really a bug but a missing feature :)

It seems not to be possible to use authentication when connecting to Elasticsearch, or is it?

The call `client = Elasticsearch([{'host': process.host, 'port': process.port}])` doesn't allow for it.

Can we have something like please? :)

```
client = Elasticsearch([{'host': process.host, 'port': process.port}], 
                       http_auth=(config['elasticsearch_user'], config['elasticsearch_pwd']))
```

**Comment by fizyk** [comment]
You'd also need to start the elasticseach process with some option to require password, and a password configured there as well 🤔

**Comment by konstantinmiller** [comment]
In our setup, we are using an Elastic Cloud instance to run integration tests on the build agent. That instance is protected and we need to use credentials to access it.

**Comment by fizyk** [comment]
@konstantinmiller I agree, however I'd need also to be able to start the server so the basic usecase with starting elasticsearch service with authentication and using it in the client would also be covered.

If you want you can work on this, as I do not know when I'll be able to add it.
